src/components/catalog-parameters/catalog-parameters.controller.ts

```diff
@@ -74,6 +81,84 @@ export class CatalogParametersController implements angular.IController {
     }
   }
 
+  private setupFormDependencies() {
+    _.each(_.get(this.ctrl.parameterSchema, 'dependencies'), (dependency: any, key: string) => {
+      if (_.isArray(dependency.oneOf)) {
+        // Only support 'oneOf' at this time, not 'allOf' or 'anyOf'
+        // See https://github.com/mozilla-services/react-jsonschema-form#dynamic for example 'dependencies' object
+        _.each(dependency.oneOf, (schema) => {
+          this.mapDependencyCondition(schema, key);
+        });
+      } else {
+        // simple nested dependency
+        this.mapDependencyCondition(dependency, key);
+      }
+    });
+  }
+
+  private mapDependencyCondition(schema, key) {
+    // Get the fields to show if the dependency is valid
+    let requiredFields = _.get(schema, 'required');
+    if (!_.isArray(requiredFields)) {
+      return;
+    }
+    _.each(requiredFields, (requiredField: any) => {
+      let fieldIndex = _.findIndex(this.ctrl.parameterForm, (field: any) => {
+        if (_.isObject(field)) {
+          return requiredFields.indexOf(field.key) > -1;
+        } else {
+          return requiredFields.indexOf(field) > -1;
+        }
+      });
+
+      // Only continue if the field is in the form defintion
+      if (fieldIndex > -1) {
+        // Normalise the field object
+        let field = this.ctrl.parameterForm[fieldIndex];
+        if (_.isString(field)) {
+          field = {
+            key: field
+          };
+        }
+
+        // Check if the required field is already in the schema properties.
+        // Add it if not.
+        if (_.isUndefined(this.ctrl.parameterSchema.properties[field.key])) {
+          this.ctrl.parameterSchema.properties[field.key] = schema.properties[field.key];
+        }
+
+        // Set the angular 'condition'
+        let enumMatches = _.get(schema, 'properties.' + key + '.enum');
+        if (_.isUndefined(enumMatches)) {
+          // Nothing specific to match against, just add a truthy condition
+          field.condition = CatalogParametersController.CONDITIONS.fieldTruthy({
+            key: key
+          });
+        } else if (enumMatches.length > 0) {
+          field.condition = CatalogParametersController.CONDITIONS.fieldEqual({
+            key: key,
+            test: this.escapeIfString(enumMatches[0])
+          });
+          // Allow matching with any other strings in the array (if there are multiple)
+          for (var oi=1, ol=enumMatches.length; oi<ol; oi++) {
+            field.condition += CatalogParametersController.CONDITIONS.oneOf + CatalogParametersController.CONDITIONS.fieldEqual({
+              key: key,
+              test: this.escapeIfString(enumMatches[oi])
+            });
+          }
+        }
+        this.ctrl.parameterForm.splice(fieldIndex, 1, field);
+      }
+    });
+  }
+
+  private escapeIfString(value) {
+    if (_.isString(value)) {
+      return '"' + value + '"';
```

Comment:
OK, so if I understand correctly this is escaping the string to use in an Angular expression. If the string contains an `"` character, could I break out an execute arbitrary expressions?

Should we just use `JSON.stringify` on all values?

What types can `value` be?

Reply:
> OK, so if I understand correctly this is escaping the string to use in an Angular expression. If the string contains an " character, could I break out an execute arbitrary expressions?

I have tested a value with quotes in it (in the serviceplan definition), and they are escaped before they get to this point.
However, ...

> Should we just use JSON.stringify on all values?

... this makes a lot of sense and will definitely prevent arbitrary expressions.
I'll change it to use this.

> 
> What types can value be?

String, number, boolean.